src/mcts/params.cc

```diff
@@ -487,6 +553,74 @@ SearchParams::SearchParams(const OptionsDict& options)
           options.Get<int>(kMaxCollisionVisitsScalingEndId)),
       kMaxCollisionVisitsScalingPower(
           options.Get<float>(kMaxCollisionVisitsScalingPowerId)) {
+  // Calculate ratio and diff for WDL conversion from the contempt settings,
+  // unless they are changed from their default values explicitly.
+  kWDLRescaleRatio = options.Get<float>(kWDLRescaleRatioId);
```

Comment:
Move this function body to a separate function, that way it will be possible to keep kWDLRescaleRatio and kWDLRescaleDiff const.

It may even make sense to group all the related WDL params in a struct, and make the entire struct a member variable of SearchParams.  Something like:

=== In params.h:
```
class SearchParams {
public:
   struct WDLRescaleParams {
       float ratio;
       float diff;
       ...
   };
   const WDLRescaleParams& GetWDLRescaleParams() const { return kWdlRescaleParams; }
private:
   ...
   const WDLRescaleParams kWdlRescaleParams;
}
```

=== In params.cc
```
namespace {
   WDLRescaleParams BuildWDLRescaleParams(const OptionsDict& options) {
      WDLRescaleParams pars;
      /// extract
      return pars;
   }
}  // namespace

...
SearchParams::SearchParams(const OptionsDict& options)
   ...
   kWdlRescaleParams(BuildWDLRescaleParams(options)) {}
```

Reply:
This sounds like a very sensible suggestion indeed. I might however require some assistance with the exact implementation, especially with where to put the formulas.